### DISCUSSION TITLE
Browse YouTube Music

### DISCUSSION THREAD

**Discussion by findleybc** [BimmerGestalt/AAIdrive#525] [General]
I am unable to browse YouTube Music in AAidrive. When playing in standard Bluetooth, I get the Home, Library etc. per the picture attached so I assume that YouTube Music is capable. There is no folder icon in the app for YouTube either.  

![PXL_20220208_204726516](https://user-images.githubusercontent.com/21024243/153077848-72d59104-72d3-4fb2-921f-f780b08f34d4.jpg)
![PXL_20220208_204857772](https://user-images.githubusercontent.com/21024243/153077860-7ba72a15-c97c-4b0c-b1be-a18cb7e2409a.jpg)

**Comment by hufman** [comment]
You are correct, Youtube Music does implement the browsing protocol! However, it has an internal allowlist (only allowing apps as Android Auto and its simulators, the Bluetooth stack, Assistant, Clock, Google Maps, and Waze), and there is no easy way to get around this limitation. Youtube Music itself must be changed to allow other apps to connect to control and browse the app.

**Reply by findleybc** [reply]
That's too bad. Can the app switch the interface to the Bluetooth device and even better, the browse screen? At least there would be a couple of clicks less than doing it manually in order to browse.  If my programming skills were any better I would give it a try but I really don't know where to start. 
This app is still really useful. Especially the notifications. Album art is the reason I use it for YTM I just switch to the Bluetooth browser and once I have a play list, switch back. 
Thanks

**Reply by hufman** [reply]
Unfortunately, the protocol doesn't allow an app to trigger global window changes like that, except to bring itself to the foreground (sometimes). Your workaround is the only way, as far as I know.

**Comment by findleybc** [comment]
Would this Python API be of any help? 
https://github.com/sigma67/ytmusicapi

**Reply by bogdan-calapod** [reply]
That's for emulating a YouTube Music client, not for controlling a Youtube Music instance like Spotify does.

Currently there's no way to do that AFAIK 😕
